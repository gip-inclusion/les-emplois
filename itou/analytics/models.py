import uuid

from django.db import models
from django.utils import timezone

from itou.common_apps.address.departments import DEPARTMENTS
from itou.users.enums import UserKind


class DatumCode(models.TextChoices):
    # Employee record - Base
    EMPLOYEE_RECORD_COUNT = "ER-001", "FS totales"
    EMPLOYEE_RECORD_DELETED = "ER-002", "FS (probablement) supprimées"
    # Employee record - Lifecycle
    EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE = "ER-101", "FS intégrées (0000) au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE = "ER-102", "FS avec une erreur au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE = "ER-102-3436", "FS avec une erreur 3436 au premier retour"
    EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR = "ER-103", "FS ayant eu au moins un retour en erreur"
    # Approval - Base
    APPROVAL_COUNT = "AP-001", "PASS IAE total"
    APPROVAL_CANCELLED = "AP-002", "PASS IAE annulés"
    # Approval - PE notification cycle
    APPROVAL_PE_NOTIFY_SUCCESS = "AP-101", "PASS IAE synchronisés avec succès avec pole emploi"
    APPROVAL_PE_NOTIFY_PENDING = "AP-102", "PASS IAE en attente de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_ERROR = "AP-103", "PASS IAE en erreur de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_READY = "AP-104", "PASS IAE prêts à être synchronisés avec pole emploi"
    # Users
    USER_COUNT = "US-001", "Nombre d'utilisateurs"
    USER_JOB_SEEKER_COUNT = "US-011", "Nombre de demandeurs d'emploi"
    USER_PRESCRIBER_COUNT = "US-012", "Nombre de prescripteurs"
    USER_EMPLOYER_COUNT = "US-013", "Nombre d'employeurs"
    USER_LABOR_INSPECTOR_COUNT = "US-014", "Nombre d'inspecteurs du travail"
    USER_ITOU_STAFF_COUNT = "US-015", "Nombre d'administrateurs"
    # API usage
    API_TOTAL_CALLS = "API-001", "API : total d'appels reçus"
    API_TOTAL_UV = "API-002", "API : total de visiteurs uniques"
    API_TOTAL_CALLS_CANDIDATS = "API-003", "API candidats : total d'appels reçus"
    API_TOTAL_UV_CANDIDATS = "API-004", "API candidats : total de visiteurs uniques"
    API_TOTAL_CALLS_GEIQ = "API-005", "API GEIQ : total d'appels reçus"
    API_TOTAL_UV_GEIQ = "API-006", "API GEIQ : total de visiteurs uniques"
    API_TOTAL_CALLS_ER = "API-007", "API FS : total d'appels reçus"
    API_TOTAL_UV_ER = "API-008", "API FS : total de visiteurs uniques"
    API_TOTAL_CALLS_MARCHE = "API-009", "API Le marché : total d'appels reçus"
    API_TOTAL_CALLS_SIAES = "API-010", "API siaes : total d'appels reçus"
    API_TOTAL_UV_SIAES = "API-011", "API siaes : total de visiteurs uniques (par adresse IP)"
    API_TOTAL_CALLS_STRUCTURES = "API-012", "API structures : total d'appels reçus"


class Datum(models.Model):
    """Store an aggregated `value` of the `code` data point for the specified `bucket`."""

    id = models.UUIDField(primary_key=True, default=uuid.uuid4)

    code = models.TextField(choices=DatumCode.choices)
    bucket = models.TextField()
    value = models.IntegerField()  # Integer offers the best balance between range, storage size, and performance

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "data"
        unique_together = ["code", "bucket"]
        indexes = [models.Index(fields=["measured_at", "code"])]


class StatsDashboardVisit(models.Model):
    dashboard_id = models.IntegerField(verbose_name="ID tableau de bord Metabase")
    dashboard_name = models.TextField(verbose_name="nom de la vue du tableau de bord")
    department = models.CharField(verbose_name="département", choices=DEPARTMENTS.items(), max_length=3, null=True)
    region = models.TextField(verbose_name="région", null=True)
    current_company_id = models.IntegerField(verbose_name="ID entreprise courante", null=True)
    current_prescriber_organization_id = models.IntegerField(
        verbose_name="ID organisation prescriptrice courante", null=True
    )
    current_institution_id = models.IntegerField(verbose_name="ID institution courante", null=True)
    user_kind = models.TextField(verbose_name="type d'utilisateur", choices=UserKind.choices)
    user_id = models.IntegerField(verbose_name="ID utilisateur")

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "visite de tableau de bord"
