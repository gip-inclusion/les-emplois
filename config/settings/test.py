import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
