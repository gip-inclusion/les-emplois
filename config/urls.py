from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
