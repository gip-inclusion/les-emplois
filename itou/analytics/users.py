from django.db.models import Count

from itou.users import enums as users_enums, models as users_models

from . import models


def collect_analytics_data(before):
    counts = {
        info["kind"]: info["count"]
        for info in users_models.User.objects.values("kind").annotate(count=Count("pk")).values("kind", "count")
    }
    return {
        models.DatumCode.USER_COUNT: sum(counts.values()),
        models.DatumCode.USER_JOB_SEEKER_COUNT: counts.get(users_enums.KIND_JOB_SEEKER, 0),
        models.DatumCode.USER_PRESCRIBER_COUNT: counts.get(users_enums.KIND_PRESCRIBER, 0),
        models.DatumCode.USER_EMPLOYER_COUNT: counts.get(users_enums.KIND_EMPLOYER, 0),
        models.DatumCode.USER_LABOR_INSPECTOR_COUNT: counts.get(users_enums.KIND_LABOR_INSPECTOR, 0),
        models.DatumCode.USER_ITOU_STAFF_COUNT: counts.get(users_enums.KIND_ITOU_STAFF, 0),
    }
