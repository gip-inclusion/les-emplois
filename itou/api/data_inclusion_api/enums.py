import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
