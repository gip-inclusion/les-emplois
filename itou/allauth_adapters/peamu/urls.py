from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
