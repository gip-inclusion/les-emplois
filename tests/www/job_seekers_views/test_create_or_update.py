import datetime

import pytest
from django.urls import reverse
from pytest_django.asserts import assertContains, assertRedirects

from itou.asp.models import Commune, Country
from itou.users.enums import Title
from tests.companies.factories import CompanyFactory
from tests.users.factories import (
    JobSeekerFactory,
)
from tests.utils.test import KNOWN_SESSION_KEYS


class TestCreateForJobSeeker:
    def test_check_nir_with_session(self, client):
        company = CompanyFactory(with_membership=True)
        user = JobSeekerFactory(jobseeker_profile__birthdate=None, jobseeker_profile__nir="")
        reset_url = reverse("companies_views:card", kwargs={"siae_id": company.pk})
        client.force_login(user)

        # Init session
        start_url = reverse("apply:start", kwargs={"company_pk": company.pk})
        client.get(start_url)
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        response = client.get(
            reverse("job_seekers_views:check_nir_for_job_seeker", kwargs={"session_uuid": job_seeker_session_name})
        )

        assertContains(response, company.display_name)
        assertContains(
            response,
            f"""
        <a href="{reset_url}" class="btn btn-link btn-ico ps-lg-0 w-100 w-lg-auto"
           aria-label="Annuler la saisie de ce formulaire">
            <i class="ri-close-line ri-lg" aria-hidden="true"></i>
            <span>Annuler</span>
        </a>""",
            html=True,
        )

    def test_cannot_check_nir_if_already_set(self, client):
        company = CompanyFactory(with_membership=True)
        user = JobSeekerFactory(
            jobseeker_profile__birthdate=datetime.date(1994, 2, 22), jobseeker_profile__nir="194022734304328"
        )
        client.force_login(user)

        # Init session
        start_url = reverse("apply:start", kwargs={"company_pk": company.pk})
        client.get(start_url)
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        response = client.get(
            reverse("job_seekers_views:check_nir_for_job_seeker", kwargs={"session_uuid": job_seeker_session_name})
        )
        assertRedirects(
            response,
            reverse(
                "job_seekers_views:check_job_seeker_info",
                kwargs={"company_pk": company.pk, "job_seeker_public_id": user.public_id},
            ),
        )


class TestCreateForSender:
    def test_check_nir_with_session(self, client):
        company = CompanyFactory(with_membership=True)
        client.force_login(company.members.get())

        # Init session
        start_url = reverse("apply:start", kwargs={"company_pk": company.pk})
        client.get(start_url)
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        response = client.get(
            reverse("job_seekers_views:check_nir_for_sender", kwargs={"session_uuid": job_seeker_session_name})
        )

        assertContains(response, company.display_name)

    @pytest.mark.parametrize(
        "born_in_france", [pytest.param(True, id="born_in_france"), pytest.param(False, id="born_outside_france")]
    )
    def test_create_step_1(self, born_in_france, client):
        company = CompanyFactory(with_membership=True)
        client.force_login(company.members.get())

        # Init session
        client.get(reverse("apply:start", kwargs={"company_pk": company.pk}))
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        birthdate = datetime.date(1911, 11, 1)
        response = client.post(
            reverse(
                "job_seekers_views:create_job_seeker_step_1_for_sender",
                kwargs={"session_uuid": job_seeker_session_name},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "nir": "111116411111144",
                "birthdate": birthdate.isoformat(),
                **(
                    {
                        "birth_place": Commune.objects.by_insee_code_and_period("64483", birthdate).pk,
                        "birth_country": Country.objects.get(code=Country.INSEE_CODE_FRANCE).pk,
                    }
                    if born_in_france
                    else {"birth_country": Country.objects.exclude(code=Country.INSEE_CODE_FRANCE).first().pk}
                ),
            },
        )
        assertRedirects(
            response,
            reverse(
                "job_seekers_views:create_job_seeker_step_2_for_sender",
                kwargs={"session_uuid": job_seeker_session_name},
            ),
        )

    def test_birth_country_not_france_and_birthplace(self, client):
        company = CompanyFactory(with_membership=True)
        client.force_login(company.members.get())

        # Init session
        client.get(reverse("apply:start", kwargs={"company_pk": company.pk}))
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        birthdate = datetime.date(1911, 11, 1)
        response = client.post(
            reverse(
                "job_seekers_views:create_job_seeker_step_1_for_sender",
                kwargs={"session_uuid": job_seeker_session_name},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "nir": "111116411111144",
                "birthdate": birthdate.isoformat(),
                "birth_place": Commune.objects.by_insee_code_and_period("64483", birthdate).pk,
                "birth_country": Country.objects.exclude(code=Country.INSEE_CODE_FRANCE).order_by("?").first().pk,
            },
        )
        assertContains(
            response,
            """
            <div class="alert alert-danger" role="alert">
                Il n'est pas possible de saisir une commune de naissance hors de France.
            </div>""",
            html=True,
            count=1,
        )

    def test_birth_country_france_and_no_birthplace(self, client):
        company = CompanyFactory(with_membership=True)
        client.force_login(company.members.get())

        # Init session
        client.get(reverse("apply:start", kwargs={"company_pk": company.pk}))
        [job_seeker_session_name] = [k for k in client.session.keys() if k not in KNOWN_SESSION_KEYS]

        response = client.post(
            reverse(
                "job_seekers_views:create_job_seeker_step_1_for_sender",
                kwargs={"session_uuid": job_seeker_session_name},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "nir": "111111111111120",
                "birthdate": "1911-11-01",
                # No birth_place
                "birth_country": Country.objects.get(code=Country.INSEE_CODE_FRANCE).pk,
            },
        )
        assertContains(
            response,
            """
            <div class="alert alert-danger" role="alert">
            Si le pays de naissance est la France, la commune de naissance est obligatoire.
            </div>""",
            html=True,
            count=1,
        )


class TestUpdateJobSeekerStep1:
    @pytest.mark.parametrize(
        "born_in_france", [pytest.param(True, id="born_in_france"), pytest.param(False, id="born_outside_france")]
    )
    def test_create_step_1(self, born_in_france, client):
        company = CompanyFactory(with_membership=True)
        user = company.members.get()
        job_seeker = JobSeekerFactory(created_by=user, title=Title.M, jobseeker_profile__nir="111116411111144")
        client.force_login(user)

        birthdate = datetime.date(1911, 11, 1)
        response = client.post(
            reverse(
                "job_seekers_views:update_job_seeker_step_1",
                kwargs={"job_seeker_public_id": job_seeker.public_id, "company_pk": company.pk},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "birthdate": birthdate.isoformat(),
                **(
                    {
                        "birth_place": Commune.objects.by_insee_code_and_period("64483", birthdate).pk,
                        "birth_country": Country.objects.get(code=Country.INSEE_CODE_FRANCE).pk,
                    }
                    if born_in_france
                    else {"birth_country": Country.objects.exclude(code=Country.INSEE_CODE_FRANCE).first().pk}
                ),
            },
        )
        assertRedirects(
            response,
            reverse(
                "job_seekers_views:update_job_seeker_step_2",
                kwargs={"job_seeker_public_id": job_seeker.public_id, "company_pk": company.pk},
            ),
        )

    def test_birth_country_not_france_and_birthplace(self, client):
        company = CompanyFactory(with_membership=True)
        user = company.members.get()
        job_seeker = JobSeekerFactory(created_by=user, title=Title.M, jobseeker_profile__nir="111116411111144")
        client.force_login(user)

        birthdate = datetime.date(1911, 11, 1)
        response = client.post(
            reverse(
                "job_seekers_views:update_job_seeker_step_1",
                kwargs={"job_seeker_public_id": job_seeker.public_id, "company_pk": company.pk},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "birthdate": birthdate.isoformat(),
                "birth_place": Commune.objects.by_insee_code_and_period("64483", birthdate).pk,
                "birth_country": Country.objects.exclude(code=Country.INSEE_CODE_FRANCE).order_by("?").first().pk,
            },
        )
        assertContains(
            response,
            """
            <div class="alert alert-danger" role="alert">
            Il n'est pas possible de saisir une commune de naissance hors de France.
            </div>""",
            html=True,
            count=1,
        )

    def test_birth_country_france_and_no_birthplace(self, client):
        company = CompanyFactory(with_membership=True)
        user = company.members.get()
        job_seeker = JobSeekerFactory(created_by=user, title=Title.M, jobseeker_profile__nir="111116411111144")
        client.force_login(user)

        response = client.post(
            reverse(
                "job_seekers_views:update_job_seeker_step_1",
                kwargs={"job_seeker_public_id": job_seeker.public_id, "company_pk": company.pk},
            ),
            {
                "title": Title.M,
                "first_name": "Manuel",
                "last_name": "Calavera",
                "birthdate": "1911-11-01",
                # No birth_place
                "birth_country": Country.objects.get(code=Country.INSEE_CODE_FRANCE).pk,
            },
        )
        assertContains(
            response,
            """
            <div class="alert alert-danger" role="alert">
            Si le pays de naissance est la France, la commune de naissance est obligatoire.
            </div>""",
            html=True,
            count=1,
        )
