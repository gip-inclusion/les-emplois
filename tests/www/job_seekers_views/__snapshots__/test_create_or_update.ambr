# serializer version: 1
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_email[in_list_application]
  '''
  <div aria-labelledby="email-confirmation-label" aria-modal="true" class="modal" id="email-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="email-confirmation-label">Email existant</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  L'adresse jane.doe@test.local est associée au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                              
                          </div>
                          <div class="modal-footer">
  
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_email[in_list_organization]
  '''
  <div aria-labelledby="email-confirmation-label" aria-modal="true" class="modal" id="email-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="email-confirmation-label">Email existant</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  L'adresse jane.doe@test.local est associée au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                              
                          </div>
                          <div class="modal-footer">
  
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_email[in_list_user]
  '''
  <div aria-labelledby="email-confirmation-label" aria-modal="true" class="modal" id="email-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="email-confirmation-label">Email existant</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  L'adresse jane.doe@test.local est associée au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                              
                          </div>
                          <div class="modal-footer">
  
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_email[not_in_list]
  '''
  <div aria-labelledby="email-confirmation-label" aria-modal="true" class="modal" id="email-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="email-confirmation-label">Email existant</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  L'adresse jane.doe@test.local est associée au compte de <b>Jane DOE</b>.
                              </p>
                              
                              <p>
                                  
                                      Le compte de ce candidat sera ajouté à votre liste une fois que vous aurez postulé pour lui.
                                  
                              </p>
                              
                          </div>
                          <div class="modal-footer">
  
                              
                                  
                                      
                                      <button class="btn btn-sm btn-outline-primary" name="cancel" type="submit" value="1">Ce n'est pas mon candidat</button>
                                  
                                  
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Continuer</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_nir[in_list_application]
  '''
  <div aria-labelledby="nir-confirmation-label" aria-modal="true" class="modal" id="nir-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="nir-confirmation-label">Utilisateur trouvé</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  Le numéro <span>2</span><span class="ms-1">90</span><span class="ms-1">01</span><span class="ms-1">01</span><span class="ms-1">010</span><span class="ms-1">101</span><span class="ms-1">25</span> est associé au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                          </div>
                          <div class="modal-footer">
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_nir[in_list_organization]
  '''
  <div aria-labelledby="nir-confirmation-label" aria-modal="true" class="modal" id="nir-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="nir-confirmation-label">Utilisateur trouvé</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  Le numéro <span>2</span><span class="ms-1">90</span><span class="ms-1">01</span><span class="ms-1">01</span><span class="ms-1">010</span><span class="ms-1">101</span><span class="ms-1">25</span> est associé au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                          </div>
                          <div class="modal-footer">
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_nir[in_list_user]
  '''
  <div aria-labelledby="nir-confirmation-label" aria-modal="true" class="modal" id="nir-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="nir-confirmation-label">Utilisateur trouvé</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  Le numéro <span>2</span><span class="ms-1">90</span><span class="ms-1">01</span><span class="ms-1">01</span><span class="ms-1">010</span><span class="ms-1">101</span><span class="ms-1">25</span> est associé au compte de <b>Jane DOE</b> figurant dans votre liste de candidats.
                              </p>
                              
                              <p>
                                  
                              </p>
                          </div>
                          <div class="modal-footer">
                              
                                  
                                  <a class="btn btn-sm btn-outline-primary" href="/job-seekers/details/7614fc4b-aef9-4694-ab17-12324300180a" role="button">Consulter le profil</a>
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Postuler</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
# name: TestStandaloneCreateAsPrescriber.test_standalone_creation_as_prescriber_existing_nir[not_in_list]
  '''
  <div aria-labelledby="nir-confirmation-label" aria-modal="true" class="modal" id="nir-confirmation-modal" role="dialog" tabindex="-1">
                  <div class="modal-dialog modal-dialog-centered">
                      <div class="modal-content">
                          <div class="modal-header">
                              <h3 class="modal-title" id="nir-confirmation-label">Utilisateur trouvé</h3>
                              <button aria-label="Fermer" class="btn-close" data-bs-dismiss="modal" type="button"></button>
                          </div>
                          <div class="modal-body">
                              
                              <p>
                                  Le numéro <span>2</span><span class="ms-1">90</span><span class="ms-1">01</span><span class="ms-1">01</span><span class="ms-1">010</span><span class="ms-1">101</span><span class="ms-1">25</span> est associé au compte de <b>Jane DOE</b>.
                              </p>
                              
                              <p>
                                  
                                      Le compte de ce candidat sera ajouté à votre liste une fois que vous aurez
                                      validé son éligibilité à l’IAE ou
                                      postulé pour lui.
                                  
                              </p>
                          </div>
                          <div class="modal-footer">
                              
                                  
                                      
                                      <button class="btn btn-sm btn-outline-primary" name="cancel" type="submit" value="1">Ce n'est pas mon candidat</button>
                                  
                                  
                                  <button class="btn btn-sm btn-primary" name="confirm" type="submit" value="1">Continuer</button>
                              
                          </div>
                      </div>
                  </div>
              </div>
  '''
# ---
